src/Microsoft.OData.Core/Json/PooledByteBufferWriter.cs

```diff
@@ -0,0 +1,221 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="PooledByteBufferWriter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP3_1_OR_GREATER
+using System;
+using System.Buffers;
+using System.Diagnostics;
+
+namespace Microsoft.OData.Json
+{
+    // This class has been adapated from https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/System/Text/Json/PooledByteBufferWriter.cs
+    /// <summary>
+    /// An implementation of <see cref="IBufferWriter{byte}"/> that rents
+    /// buffers from an array pool instead of allocating a new array every time.
+    /// </summary>
+    internal sealed class PooledByteBufferWriter : IBufferWriter<byte>, IDisposable
+    {
+        // This class allows two possible configurations: if rentedBuffer is not null then
+        // it can be used as an IBufferWriter and holds a buffer that should eventually be
+        // returned to the shared pool. If rentedBuffer is null, then the instance is in a
+        // cleared/disposed state and it must re-rent a buffer before it can be used again.
+        private byte[] rentedBuffer;
+        private int index;
+
+        private const int MinimumBufferSize = 256;
+
+        // Value copied from Array.MaxLength in https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Array.cs
+        public const int MaximumBufferSize = 0X7FFFFFC7;
+
+        private PooledByteBufferWriter()

```

Comment:
Why do we need this constructor? A private constructor is used to prevent the automatic generation of a public parameterless constructor but in your case there's already a public constructor that takes a parameter.

Reply:
Good point. It's not needed. Must be some artifact of referenced code.